lib/models/addon-discovery.js

```diff
@@ -11,7 +11,10 @@ const path = require('path');
 const resolve = require('resolve');
 const findup = require('find-up');
 const heimdall = require('heimdalljs');
-
+const PACKAGE_CACHE = Object.create(null);
```

Comment:
unsure, as when `require` is invoked the result is frozen. require'ing again with a different node_module topology is already not going to work.

If you can provide some legitimate scenarios where this is expected to work?